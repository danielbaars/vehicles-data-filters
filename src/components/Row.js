import React from 'react';
import PropTypes from 'prop-types';

const Row = ({ extraClasses, children }) => {
  return (
    <div className='grid-container'>
      <div className={`grid-x grid-margin-x ${extraClasses}`}>
        {children}
      </div>
    </div>
  );
};

export default Row;

Row.propTypes = {
  extraClasses: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Row.defaultProps = {
  extraClasses: '',
};
