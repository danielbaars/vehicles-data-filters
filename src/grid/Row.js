import React from 'react';
import PropTypes from 'prop-types';

const RowContent = ({ extraClasses, children }) => {
  return (
    <div className={`grid-x grid-margin-x ${extraClasses}`}>
      {children}
    </div>
  );
};

const Row = ({ extraClasses, container, children }) => {
  if (container) {
    return (
      <div className='grid-container'>
        <RowContent extraClasses={extraClasses}>{children}</RowContent>
      </div>
    );
  }
  return <RowContent extraClasses={extraClasses}>{children}</RowContent>;
};

export default Row;

RowContent.propTypes = {
  extraClasses: PropTypes.string,
  children: PropTypes.node.isRequired,
};

RowContent.defaultProps = {
  extraClasses: '',
};

Row.propTypes = {
  extraClasses: PropTypes.string,
  container: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

Row.defaultProps = {
  extraClasses: '',
  container: false,
};
