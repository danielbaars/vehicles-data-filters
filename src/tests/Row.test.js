import React from 'react';
import renderer from 'react-test-renderer';

import Row from '../components/Row';

describe('Row', () => {
  const component = <Row extraClasses='class-name'><div>Some Child</div></Row>;
  it('renders and matches our snapshot', () => {
    const sComponent = renderer.create(component);
    const tree = sComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
