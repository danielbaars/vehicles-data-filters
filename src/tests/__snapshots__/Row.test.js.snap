// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders and matches our snapshot 1`] = `
<div
  className="grid-x grid-margin-x class-name"
>
  <div>
    Some Child
  </div>
</div>
`;
